src/retransmit_stage.rs

```diff
@@ -37,13 +38,47 @@ fn retransmit(
             .to_owned(),
     );
 
-    for b in &mut dq {
-        ClusterInfo::retransmit(&cluster_info, b, sock)?;
+    let peers = cluster_info.read().unwrap().sorted_tvu_peers(bank);
+    let my_id = cluster_info.read().unwrap().id();
+    //calc num_layers and num_neighborhoods using the total number of nodes
+    let (num_layers, layer_indices) = ClusterInfo::describe_data_plane(
+        peers.len(),
+        DATA_PLANE_FANOUT,
+        NEIGHBORHOOD_SIZE,
+        GROW_LAYER_CAPACITY,
+    );
+    if num_layers <= 1 {
+        /* single layer data plane */
+        for b in &mut dq {
+            ClusterInfo::retransmit(&cluster_info, b, sock)?;
+        }
+    } else {
+        //find my index (my ix is the same as the first node with smaller stake)
+        let my_index = peers
+            .iter()
+            .position(|ci| bank.get_stake(&ci.id) <= bank.get_stake(&my_id));
+        //find my layer
+        let locality = ClusterInfo::localize(
+            &layer_indices,
+            NEIGHBORHOOD_SIZE,
+            my_index.unwrap_or(peers.len() - 1),
+        );
+        let mut retransmit_peers =
+            peers[locality.neighbor_bounds.0..locality.neighbor_bounds.1].to_vec();
+        locality.child_layer_peers.iter().cloned().for_each(|ix| {
+            if let Some(peer) = peers.get(ix) {
+                retransmit_peers.push(peer.clone());
+            }
+        });
+
+        for b in &mut dq {
+            ClusterInfo::retransmit_to(&cluster_info, retransmit_peers.to_vec(), b, sock)?;
```

Comment:
retransmit_peers.to_vec() necessary?  not already a vector?

Reply:
Yeah this looks weird. I'll check it out.